flink-runtime/src/main/java/org/apache/flink/runtime/scheduler/SimpleExecutionSlotAllocator.java

```diff
@@ -74,9 +74,10 @@ public class SimpleExecutionSlotAllocator implements ExecutionSlotAllocator {
     }
 
     @Override
-    public List<ExecutionSlotAssignment> allocateSlotsFor(
+    public Map<ExecutionAttemptID, ExecutionSlotAssignment> allocateSlotsFor(
             List<ExecutionAttemptID> executionAttemptIds) {
-        List<ExecutionSlotAssignment> result = new ArrayList<>(executionAttemptIds.size());
+        Map<ExecutionAttemptID, ExecutionSlotAssignment> result =
+                new HashMap<>(executionAttemptIds.size());
```

Comment:
Setting initial capacity for maps should be avoided, because it usually does not help.